lib/puppet/provider/package/gem.rb

```diff
@@ -122,7 +122,12 @@ def query
   end
 
   def uninstall
-    gemcmd "uninstall", "-x", "-a", resource[:name]
+    command = [command(:gemcmd), "uninstall"]
```

Comment:
It is unfortunate that you have to do this... When you declare a command in your provider, puppet automatically creates a method, e.g. `gemcmd`, which ultimately calls into `Puppet::Util::Execution.execute` to execute the command. Unfortunately, it appears `sensu_gem` is able to override the value of `command(:gem)`, but not the path used in `gemcmd`. So it's surprising when you use one vs the other:

```
0 ~/work/puppet (master)  $ bundle exec irb
...
irb(main):012:0> require 'puppet'
=> true
irb(main):013:0> Puppet::Type.type(:package).provide :sensu_gem, :parent => :gem do
irb(main):014:1*   desc "Sensu Embedded Ruby Gem support. If a URL is passed via `source`, then
irb(main):015:1"     that URL is used as the remote gem repository; if a source is present but is
irb(main):016:1"     not a valid URL, it will be interpreted as the path to a local gem file.  If
irb(main):017:1"     source is not present at all, the gem will be installed from the default gem
irb(main):018:1"     repositories."
irb(main):019:1>
irb(main):020:1*   has_feature :versionable, :install_options
irb(main):021:1>
irb(main):022:1*   commands :gemcmd => "/opt/sensu/embedded/bin/gem"
irb(main):023:1> end
=> Puppet::Type::Package::ProviderSensu_gem
...
irb(main):028:0> provider = Puppet::Type.type(:package).provider(:sensu_gem)
=> Puppet::Type::Package::ProviderSensu_gem
irb(main):029:0> provider.command(:gemcmd)
=> nil
```

This is because I don't have `/opt/sensu/embedded/bin/gem`, so this is returning the overridden value as expected. But

```
irb(main):030:0> provider.gemcmd
Puppet::ExecutionFailure: Execution of '/Users/josh/.rbenv/versions/2.1.5/bin/gem' returned 1: RubyGems is a sophisticated package manager for Ruby.
```

Is executing using the gem command from the base class.

So I'm :+1: with the changes, just sad puppet is not consistent in how commands are overridden.